File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Fee.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "fees", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "fees_unique", columnNames = {"fee_name"})
})
public class Fee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "fee_id", nullable = false)
    private Long id;

    @Column(name = "fee_percent")
    private Double feePercent;

    @Column(name = "fixed_fee")
    private Double fixedFee;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id")
    private Unit unit;

    @NotNull
    @Column(name = "fee_name", nullable = false, length = Integer.MAX_VALUE)
    private String feeName;

}

File: umbrella-insurance-react-ui/src/models/users/fees/v1/Fee.ts
import { Unit } from "../../../units/v1/Unit";

export class Fee {
    id?: number;
    feePercent?: number;
    fixedFee?: number;
    unit?: Unit;//currency
    feeName?: string;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.feePercent) {
                this.feePercent = obj.feePercent;
            }
            if(obj?.fixedFee) {
                this.fixedFee = obj.fixedFee;
            }
            if(obj?.unit) {
                this.unit = new Unit(obj.unit);
            }
            if(obj?.feeName) {
                this.feeName = obj.feeName;
            }
        }
    }
}